samples/charts/category-chart/line-chart-with-annotations/App.razor

```diff
@@ -0,0 +1,130 @@
+            </label>
+            <label class="igOptions-item">
+                <input type="checkbox" @onchange="OnFinalValuesCheckboxChange" checked="@FinalValuesVisible" /> Final Values
+            </label>
+            <label class="igOptions-item">
+                <input type="checkbox" @onchange="OnMarkersVisibleCheckBoxChange" checked="@MarkersVisible" /> Markers
+            </label>
+        </div>
+    </div>
+
+    <div class="igComponent">
+        @if (DataSource != null)
+        {
+            <CategoryChart Height="100%" Width="100%"
+                           @ref="CategoryChartRef"
+                           DataSource="DataSource"
+                           ChartType="CategoryChartType.Line"
+                           Subtitle="Renewable Electricity Generated"
+                           YAxisTitle="TWh"
+                           Thickness="2"
+                           CrosshairsSnapToData="true"
+                           CrosshairsDisplayMode="@CrosshairMode"
+                           CrosshairsAnnotationEnabled="@CrosshairsVisible"
+                           FinalValueAnnotationsVisible="@FinalValuesVisible"
+                           YAxisLabelLocation="AxisLabelsLocation.OutsideRight"  
+                           CalloutsVisible="@CalloutsVisible"
+                           CalloutsYMemberPath="Value"
+                           CalloutsXMemberPath="Index"
+                           CalloutsLabelMemberPath="Label"
+                           CalloutsDataSource="@CalloutDataSource"
+                           ExcludedProperties="@(new string[] { "China", "Europe" })">
+            </CategoryChart>
+        }
+    </div>
+</div>
+
+@code {
+
+    private List<EnergyRenewableInfo>
+    DataSource = new EnergyRenewableData();
+    private List<CalloutInfo> CalloutDataSource = new CalloutData();
+
+    private CategoryChart _CategoryChart;
+    private CategoryChart CategoryChartRef
+    {
+        get { return _CategoryChart; }
+        set { _CategoryChart = value;
+            CategoryChartRef.MarkerTypes.Add(MarkerType.Circle);
+            StateHasChanged(); }
+    }
+    private bool MarkersVisible = true;
+    private bool FinalValuesVisible = true;
+    private bool CalloutsVisible = true;
+    private bool CrosshairsVisible = true;
+
+    private CrosshairsDisplayMode CrosshairMode = CrosshairsDisplayMode.Both;
+
+    private void OnMarkersVisibleCheckBoxChange(ChangeEventArgs args)
+    {
+        bool value = (bool)args.Value;
+        if (value == true)
+        {
+            CategoryChartRef.MarkerTypes.Add(MarkerType.Automatic);
+        }
+        else {
+            CategoryChartRef.MarkerTypes.Clear();
+        }
+        this.MarkersVisible = value;
+    }
+    private void OnFinalValuesCheckboxChange(ChangeEventArgs args)
+    {
+        bool value = (bool)args.Value;
+        this.FinalValuesVisible = value;
```

Comment:
replace code with just one line:
`this.FinalValuesVisible = (bool)args.Value;`

Reply:
pending